src/buttons/CreateScenarioButton/components/CreateScenarioDialog/CreateScenarioDialog.js

```diff
@@ -32,25 +32,20 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-const getCurrentScenarioRunType = (currentScenario, runTemplates) => {
-  const runTemplateId = currentScenario?.data?.runTemplateId;
-  runTemplates = runTemplates === undefined ? [] : runTemplates;
-  const runTemplate = runTemplates.find((runTemplate) => runTemplate.id === runTemplateId);
-  return runTemplate === undefined ? {} : runTemplate;
+const getCurrentScenarioRunTemplate = (currentScenario, runTemplates) => {
+  const currentRunTemplateId = currentScenario?.runTemplateId;
+  return runTemplates.find((runTemplate) => runTemplate.id === currentRunTemplateId) || null;
 };
 
-const isDialogDataValid = (scenarioName, isMaster, scenarioType, parentScenario, dataset) => {
-  const validScenarioName = scenarioName.value.length !== 0 && !scenarioName.hasError;
-  const validScenarioType = Object.keys(scenarioType).length !== 0;
-  const validParentScenario = Object.keys(parentScenario).length !== 0;
-  const validDataset = Object.keys(dataset).length !== 0;
-  let isValid;
+const isDialogDataValid = (scenarioName, scenarioNameFieldError, isMaster, scenarioType, parentScenario, dataset) => {
+  const validScenarioName = scenarioName.length !== 0 && scenarioNameFieldError == null;
```

Comment:
Using a loose equality means we accept `scenarioNameFieldError` equal to undefined? Won't we loose some error cases knowing that it's initial state is null ? :slightly_smiling_face:

Reply:
I intended to accept both values because `undefined` values should be replaced by a default error message in the `handleChangeScenarioName`, but I guess a strict equality here could help catch some dev errors earlier :+1: